atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -329,58 +334,47 @@ def resolve_signing_intent(self):
+                self.yum_repourls['noarch'].add(result_repofile)
             else:
                 for arch in arches.split():
-                    repos_by_arch[arch].append(result_repofile)
+                    self.yum_repourls[arch].add(result_repofile)
 
         # we should almost never have a None entry from composes,
         # but we can have yum_repos added, so if we do, we need to merge
         # it with all other repos.
         if self.repourls:
-            repos_by_arch[None].extend(self.repourls)
-        try:
-            noarch_repos = repos_by_arch.pop(None)
-        except KeyError:
-            pass
-        else:
-            for repos in repos_by_arch.values():
-                repos.extend(noarch_repos)
-
-        for arch, repofiles in repos_by_arch.items():
-            # OSBS2 TBD: `override_build_kwarg` is imported from build_orchestrate_build
-            override_build_kwarg(self.workflow, 'yum_repourls', repofiles, arch)
-
-        # Only set the None override if there are no other repos
-        if not repos_by_arch:
-            # OSBS2 TBD: `override_build_kwarg` is imported from build_orchestrate_build
-            override_build_kwarg(self.workflow, 'yum_repourls', noarch_repos, None)
+            self.yum_repourls['noarch'].update(self.repourls)
+        if 'noarch' in self.yum_repourls:
+            noarch_repos = self.yum_repourls.pop('noarch')
+            for arch in self.yum_repourls:
+                self.yum_repourls[arch].update(noarch_repos)
 
         # If we don't think the set of packages available from the user-supplied repourls,
         # inherited repourls, and composed repositories is complete, set the 'include_koji_repo'
         # kwarg so that the so that the 'yum_repourls' kwarg that we just set doesn't
         # result in the 'koji' plugin being omitted.
         if not self.has_complete_repos:
-            # OSBS2 TBD: `override_build_kwarg` is imported from build_orchestrate_build
-            override_build_kwarg(self.workflow, 'include_koji_repo', True)
-
-        # So that plugins like flatpak_update_dockerfile can get information about the composes
-        # OSBS2 TBD: `override_build_kwarg` is imported from build_orchestrate_build
-        override_build_kwarg(self.workflow, 'compose_ids', self.all_compose_ids)
+            self.include_koji_repo = True
 
     def make_result(self):
+        signing_intent = None
+        signing_intent_overridden = False
+        if self.compose_config:
+            signing_intent = self.compose_config.signing_intent['name']
+            signing_intent_overridden = self.compose_config.has_signing_intent_changed()
         result = {
             'composes': self.composes_info,
-            'signing_intent': self.compose_config.signing_intent['name'],
-            'signing_intent_overridden': self.compose_config.has_signing_intent_changed(),
+            'yum_repourls': self.yum_repourls,
```

Comment:
It might be good to convert these to a `dict[str, list]` rather than `dict[str, set]` in this method - plugin results will soon need to be JSON-serializable

(@tkdchen can correct me if I'm wrong)

Reply:
incorporated suggestion